### DISCUSSION TITLE
Does [DNS rewrite] support domain-ip list? Can I import list?

### DISCUSSION THREAD

**Discussion by sapipoZZZ** [AdguardTeam/AdGuardHome#5407] [Q&A]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to request a feature or enhancement and not ask a question

#### What problem are you trying to solve?
I've about 20+ HOSTS rule to setting, But I have to input in DNS rewrite page one by one. It's not convenient for me.
#### Proposed solution
I need import function in DNS rewrite page, or edit the setting file directly like hosts file in Windows / linux

**Comment by ainar-g** [comment]
I'm not quite sure what you're asking, sorry.  You can use `/etc/hosts`-like rules as-is, by just adding them as a filter file on the *Filters → DNS blocklists* page.

**Reply by ammnt** [reply]
> I'm not quite sure what you're asking, sorry. You can use `/etc/hosts`-like rules as-is, by just adding them as a filter file on the _Filters → DNS blocklists_ page.



> I'm not quite sure what you're asking, sorry. You can use `/etc/hosts`-like rules as-is, by just adding them as a filter file on the _Filters → DNS blocklists_ page.

Looks like the user want to import/export DNS rewrites as list. Just like "DNS allowlists" or "DNS blocklist"🤔

**Reply by sapipoZZZ** [reply]
Yes. 

The large number of domain name resolutions provided by the network operator in my area have problems, so I have to manually calibrate these problematic domain names through the DNS rewrite function.

The "DNS Rewrite" function currently only allows for adding or editing rules one by one, which is very inconvenient. I need a batch adding or editing function.

**Reply by fvdm** [reply]
If you mean to migrate an `/etc/hosts` file to AGH, then in _Filters_ > _Custom filtering rules_ you can directly copy/paste the contents of the file all at once.

AGH supports mixed syntax. So you can combine all kinds of rules in that textfield, including hosts-style `1.2.3.4 domain.tld` lines.

https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#etc-hosts

When it is a large amount of rules check out the [Hostlist compiler](https://github.com/AdguardTeam/HostlistCompiler). It converts the hosts file contents to adblock style in optimized form, allowing advanced options like DNS rewrites and more. You can then copy/paste the contents of the output file to the _Custom filtering rules_ textfield or place it on a webserver and configure the URL in _Filters_ > _DNS blocklists_ > _Add blocklist_ > _Add a custom list_. Even though it says 'blocklist' in the menu you can use allows and rewrites in that file too.
